sql/log_event.cc

```diff
@@ -10199,6 +10200,26 @@ int Rows_log_event::do_apply_event(Relay_log_info const *rli) {
 
 end:
   if (get_flags(STMT_END_F)) {
+    if (table != nullptr && thd->slave_thread) {
+      /*
+        Set short description of RBR event to be used by slow query log.
+       */
+      std::ostringstream log_query;
+      log_query << get_type_str() << " " << table->s->db.str << "."
+                << table->s->table_name.str;
+      char *log_query_copy =
+          static_cast<char *>(thd->alloc(log_query.str().length() + 1));
+      if (log_query_copy != nullptr) {
+        memcpy(log_query_copy, log_query.str().c_str(),
+               log_query.str().length());
+        log_query_copy[log_query.str().length()] = '\0';
+        thd->set_query(log_query_copy, log_query.str().length());
```

Comment:
Since we are calling `thd->set_query()`  before logging, there are chances that `SHOW PROCESSLIST`  on replica server may show the modified query. We need to check if this is okay.
```
mysql> show processlist;
...
+----+-----------------+-----------------+------+---------+------+--------------------------------------------------------+--------------------------------------------------------+---------+-----------+---------------+
| 16 | system user     |                 | NULL | Query   |    5 | Applying batch of row changes (write)                  | Write_rows test.t                                      |    4515 |         0 |             0 |
+----+-----------------+-----------------+------+---------+------+--------------------------------------------------------+--------------------------------------------------------+---------+-----------+---------------+
8 rows in set (0.00 sec)
```
Without the patch, (with `binlog_rows_query_log_events`=1) the server shows the actual query in the Info field of SHOW PROCESSLIST .
```
| 15 | system user     |                 | NULL | Query   |    4 | Applying batch of row changes (write)                  | INSERT INTO t VALUES (1)                               |    4291 |         0 |             0 |
```
Without the patch, (with `binlog_rows_query_log_events`=0) the server shows NULL in the Info field of SHOW PROCESSLIST .
```
| 16 | system user     |                 | NULL | Query   |    5 | Applying batch of row changes (write)                  | NULL                                                   |    4588 |         0 |             0 |
```